lib/src/main/java/com/orange/ods/compose/component/button/OdsFloatingActionButton.kt

```diff
@@ -79,31 +83,64 @@ fun OdsFloatingActionButton(
  *
  * @see androidx.compose.material.ExtendedFloatingActionButton for further information.
  *
+ * @param icon The FAB icon.
  * @param onClick callback invoked when this FAB is clicked
  * @param text The text displayed in the [OdsExtendedFloatingActionButton]
- * @param icon [Painter] of the FAB icon.
  * @param modifier [Modifier] to be applied to this FAB.
- * @param iconContentDescription [icon] content description, can be null cause there is a text.
  */
 @Composable
 @OdsComposable
 fun OdsExtendedFloatingActionButton(
+    icon: OdsFloatingActionButtonIcon,
     onClick: () -> Unit,
     text: String,
-    icon: Painter,
     modifier: Modifier = Modifier,
-    iconContentDescription: String? = null
 ) {
     ExtendedFloatingActionButton(
         onClick = onClick,
         text = { Text(text = text.uppercase(), style = OdsTheme.typography.button) },
         modifier = modifier,
         backgroundColor = OdsTheme.colors.component.floatingActionButton.background,
         contentColor = OdsTheme.colors.component.floatingActionButton.content,
-        icon = { Icon(painter = icon, contentDescription = iconContentDescription) }
+        icon = { icon.Content(modifier = Modifier.size(FabIconSize)) }
     )
 }
 
+/**
+ * A button icon in an [OdsFloatingActionButton].
+ */
+class OdsFloatingActionButtonIcon : OdsComponentIcon {
+
+    /**
+     * Creates an instance of [OdsFloatingActionButtonIcon].
+     *
+     * @param painter Painter of the icon.
+     * @param contentDescription The content description associated to this [OdsFloatingActionButtonIcon].
+     */
+    constructor(painter: Painter, contentDescription: String) : super(painter as Any, contentDescription)
+
+    /**
+     * Creates an instance of [OdsFloatingActionButtonIcon].
+     *
+     * @param imageVector Image vector of the icon.
+     * @param contentDescription The content description associated to this [OdsFloatingActionButtonIcon].
+     */
+    constructor(imageVector: ImageVector, contentDescription: String) : super(imageVector as Any, contentDescription)
+
+    /**
+     * Creates an instance of [OdsFloatingActionButtonIcon].
+     *
+     * @param bitmap Image bitmap of the icon.
+     * @param contentDescription The content description associated to this [OdsFloatingActionButtonIcon].
+     */
+    constructor(bitmap: ImageBitmap, contentDescription: String) : super(bitmap as Any, contentDescription)
+
+    @Composable
+    override fun Content(modifier: Modifier) {
+        super.Content(modifier = modifier.size(ButtonDefaults.IconSize))
+        Spacer(Modifier.size(ButtonDefaults.IconSpacing))
```

Comment:
If I am correct `Spacer` is not necessary. Size is also applied although it was already applied when calling `Content`.

Maybe this method is a copy / paste that should have been deleted?